### DISCUSSION TITLE
Accepting http and redirection to https

### DISCUSSION THREAD

**Discussion by julescole** [nuxt/nuxt#8609] [Questions (Nuxt 2)]
I've taken a look at https://github.com/nuxt-contrib/redirect-ssl and i'm trying to get nuxt to redirect http requests to https. However, once i setup the server.https variable in nuxt.config.js the app only works over https, and i simply get `ERR_EMPTY_RESPONSE` when trying to access it over http. If i remove the server.http variable, the app then works with http requests. 

Does anybody know how to resolve this, so that I can get http to redirect to https?

**Comment by jacobhq** [comment]
It might be easier to see if your hosting/domain provider supports https redirects.

**Comment by julescole** [comment]
I worked out a way in the end, by creating a new server middleware that listened on the HTTP port and redirected to the HTTPS URL.

**Comment by devlsh** [comment]
@julescole That's great - however if you don't mind me asking, what sort of deployment are you using?

As if, for example, you're using an Nginx reverse proxy, there are much better ways of redirecting than a serverMiddleware
